src/FileCache.php

```diff
@@ -325,25 +314,27 @@ private function normalizeTtl(null|int|string|DateInterval $ttl = null): ?int
     }
 
     /**
-     * Ensures that the directory is created.
+     * Ensures that the directory is existing.
      *
      * @param string $path The path to the directory.
-     *
-     * @return bool Whether the directory was created.
      */
-    private function createDirectoryIfNotExists(string $path): bool
+    private function ensureDirectory(string $path): void
     {
         if (is_dir($path)) {
-            return true;
+            return;
+        }
+
+        if (is_file($path)) {
+            throw new CacheException("Failed to create cache directory \"$path\".");
         }
 
-        $result = !is_file($path) && mkdir(directory: $path, recursive: true) && is_dir($path);
+        mkdir($path, recursive: true);
```

Comment:
Wouldn't it fail in case of race condition? `is_dir` + `mkdir` is not an atomic operation.

Reply:
It will. Race condition don't fixed in this PR. I make it next PR.